Sources/NIOSSL/SSLPrivateKey.swift

```diff
@@ -404,6 +404,19 @@ extension NIOSSLPrivateKey {
             return customKey.signatureAlgorithms
         }
     }
+
+    /// Extracts the bytes of this private key in DER format.
+    /// - Important: If the key is not an `EVP_PKEY`, an empty array will be returned.
+    /// - Returns: The DER-encoded bytes for this `EVP_PKEY` private key.
+    /// - Throws: If an error occurred while serializing the private key.
+    public func toDERBytes() throws -> [UInt8] {
+        switch self.representation {
+        case .native(let evpKey):
+            return try Self.withUnsafeDERBuffer(of: evpKey) { Array($0) }
+        case .custom(let custom):
+            return []
```

Comment:
We should allow custom private keys to handle this case, but just default them to returning the empty array.

Reply:
Also having second thoughts about the API: do you think `toDERBytes()` is more appropriate than having a `derBytes` computed property?